packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -330,41 +315,49 @@ export class JsonLdCredentialFormatService implements CredentialFormatService<Js
     })
   }
 
-  private compareCredentialSubject(
+  private verifyReceivedCredentialMatchesRequest(
     credential: W3cVerifiableCredential,
-    requestAsJson: JsonLdSignCredentialFormat
+    request: JsonLdFormatDataCredentialDetail
   ): void {
-    const request = JsonTransformer.fromJSON(requestAsJson.credential, W3cCredential)
+    // const request = JsonTransformer.fromJSON(requestAsJson.credential, W3cCredential)
 
-    if (!objectEquality(credential.credentialSubject, request.credentialSubject)) {
-      throw new AriesFrameworkError('Received credential subject does not match subject from credential request')
+    const jsonCredential = JsonTransformer.toJSON(credential)
+    delete jsonCredential.proof
+
+    // Check whether the received credential (minus the proof) matches the credential request
+    if (!areObjectsEqual(jsonCredential, request.credential)) {
```

Comment:
Can we move this check below the others? This is the most computationally heavy. If this is the main reason for invalid credentials it might be fine, but if all cases are as common as the next, it would be good to do this check at the last validation step.